### DISCUSSION TITLE
Invalid Argument (Morphology)

### DISCUSSION THREAD

**Discussion by reeve567** [route/mogrify#114] [Q&A]
I'm attempting to use the TopHat Morphology operation on two folders of images, and everything runs without the `custom("morphology")`, however when I add that line, I get this error:

```
== Compilation error in file lib/melanoma.ex ==
** (MatchError) no match of right hand side value: {"convert: invalid argument for option '-morphology': saved_images/melanoma/new_ISIC_0000002.jpg @ error/convert.c/ConvertImageCommand/2285.\r\n", 1}
    (mogrify 0.9.2) lib/mogrify.ex:80: Mogrify.create/2
    (elixir 1.14.4) lib/enum.ex:1658: Enum."-map/2-lists^map/1-0-"/2
    (elixir 1.14.4) lib/kernel/parallel_compiler.ex:340: anonymous fn/5 in Kernel.ParallelCompiler.spawn_workers/7
```

Specifically, I'm trying to use this command:

`magick man.gif   -morphology TopHat Disk  tophat_man.gif`
([Documentation](https://www.imagemagick.org/Usage/morphology/#top-hat))

Does anyone have any ideas what I could try next? This is a pretty simple project but I wanted to take the opportunity to learn some Elixir.
Any tips for debugging this would be appreciated too, since I haven't quite figured out many methods for that yet.

Code:
```
import Mogrify

kernel_size = 4.3
kernel_shape = "Octagon"

source_images = Path.wildcard("./images/*/*.jpg")

IO.puts("Listing source images: #{inspect(source_images)}")

Enum.map(source_images, fn val ->
  image = open(val)
  saved_image_path = String.replace(val, "images", "saved_images")

  IO.puts("Processing #{val}...")
  IO.puts("Saving to #{saved_image_path}...")

  image
  |> custom("morphology", "TopHat #{kernel_shape}:#{kernel_size}")
  |> create([path: saved_image_path])
end)
```

**Comment by talklittle** [comment]
Thanks for the bug report. `custom(image, "morphology", "...")` should be working now in mogrify v0.9.3

You may also need to use `save/2` instead of `create/2`. See https://github.com/elixir-mogrify/mogrify/commit/739ada03452d9c18ca83871ee316477e930471f1
